### ISSUE TITLE
`no-array-for-each`: autofix uses an already existing variable name in method chaining

### ISSUE THREAD

**Issue by noftaly** [sindresorhus/eslint-plugin-unicorn#1093]
I had this code before, with an `Array#forEach`:
```js
topics.results
  .map(topic => /* ... */)
  .filter(topic => /* ... */)
  .forEach((topic) => {
    // ...
  });
```

The new `no-array-for-each` auto-fixed it to:
```js
for (const topic of topics.results
  .map(topic => /* ... */)
  .filter(topic => /* ... */) {
    // ...
  }
```

But it used the name `topic` for the for loop, which is already used in the `.map` and `.filter` methods. This triggered the `no-shadow` rule.
I don't know if unicorn can do anything about it though, appart from extracting the array to a variable before the loop, or using a suggestion rather than a fix.

One thing I though of, is to add an option to disable this rule when using method chaining, because I think it looks better when using a `forEach` with method chaining, than using method chaining in a `for of`.

**Comment by fisker** [comment]
Don't you think manually fix  `no-shadow` is  easier than  complicated `forEach ` case?

In you  case above  I would  move `.filter` out and assign to  a  new variable.  Easy fix.

**Comment by sindresorhus** [comment]
The rule is doing the correct thing here.

**Comment by sindresorhus** [comment]
> One thing I though of, is to add an option to disable this rule when using method chaining, because I think it looks better when using a forEach with method chaining, than using method chaining in a for of.
 
👎  The solution is to just:

> In you case above I would move .filter out and assign to a new variable. Easy fix.
